catalyst/loggers/wandb.py

```diff
@@ -0,0 +1,180 @@
+
+    Args:
+        Project: Name of the project in W&B to log to.
+        name: Name of the run in W&B to log to.
+        config: Configuration Dictionary for the experiment.
+        entity: Name of W&B entity(team) to log to.
+
+    Python API examples:
+
+    .. code-block:: python
+
+        from catalyst import dl
+
+        runner = dl.SupervisedRunner()
+        runner.train(
+            ...,
+            loggers={"wandb": dl.WandbLogger(project="wandb_test", name="expeirment1")}
+        )
+
+    .. code-block:: python
+
+        from catalyst import dl
+
+        class CustomRunner(dl.IRunner):
+            # ...
+
+            def get_loggers(self):
+                return {
+                    "console": dl.ConsoleLogger(),
+                    "wandb": dl.WandbLogger(project="wandb_test", config=self.hparams)
+                }
+
+            # ...
+
+        runner = CustomRunner().run()
+
+    Config API example:
+
+    .. code-block:: yaml
+
+        loggers:
+            wandb:
+                _target_: WandbLogger
+                project: test_exp
+                name: test_run
+        ...
+
+    Hydra API example:
+
+    .. code-block:: yaml
+
+        loggers:
+            wandb:
+                _target_: catalyst.dl.WandbLogger
+                project: test_exp
+                name: test_run
+        ...
+    """
+
+    def __init__(
+        self,
+        project: str,
+        name: Optional[str] = None,
+        config: Optional[Dict] = None,
+        entity: Optional[str] = None,
+    ) -> None:
+        self.project = project
+        self.name = name
+        self.config = config
+        self.entity = entity
+        self.run = wandb.init(
+            project=self.project, name=self.name, config=self.config, entity=self.entity
+        )
+
+    def _log_metrics(self, metrics: Dict[str, float], step: int, loader_key: str, suffix=""):
+        for key, value in metrics.items():
+            self.run.log({f"{key}/{loader_key}{suffix}": value}, step=step)
+
+    def log_metrics(
+        self,
+        metrics: Dict[str, Any],
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs batch and epoch metrics to wandb."""
+        if scope == "batch":
+            metrics = {k: float(v) for k, v in metrics.items()}
+            self._log_metrics(
+                metrics=metrics, step=global_batch_step, loader_key=loader_key, suffix="/batch"
```

Comment:
```suggestion
                metrics=metrics, step=global_sample_step, loader_key=loader_key, suffix="/batch"
```
for convenient logging :)

Reply:
Okay, I tried this but as I mentioned on slack, wandb needs steps to be strictly increasing, we cannot go back to previous steps. And there's only one global step. So, doing it this way will cause it to drop values, so I'm  using `global_epoch_step` for all cases as it ensures that it's strictly increasing

Reply:
all `*_step` counters are always increasing during the Catalyst run, so that should be okay ;)

Reply:
Yeah, the individual `*_step` counters are increasing but wandb run only has one global counter. Consider a scenario where `global_step` is 3 and `sample_step` is 2. I tested with this and the values were being dropped with warnings.